Source/JavaScriptCore/runtime/PropertySlot.h

```diff
@@ -85,11 +86,11 @@ inline unsigned attributesForStructure(unsigned attributes)
     return static_cast<uint8_t>(attributes);
 }
 
-using GetValueFunc = EncodedJSValue(JIT_OPERATION_ATTRIBUTES*)(JSGlobalObject*, EncodedJSValue thisValue, PropertyName);
-using GetValueFuncWithPtr = EncodedJSValue(JIT_OPERATION_ATTRIBUTES*)(JSGlobalObject*, EncodedJSValue thisValue, PropertyName, void*);
+using GetValueFunc = TypedFunctionPtr<GetValueFuncPtrTag, EncodedJSValue, JSGlobalObject*, EncodedJSValue, PropertyName>;
+using GetValueFuncWithPtr = TypedFunctionPtr<GetValueFuncWithPtrPtrTag, EncodedJSValue, JSGlobalObject*, EncodedJSValue, PropertyName, void*>;
 
-using PutValueFunc = bool (JIT_OPERATION_ATTRIBUTES*)(JSGlobalObject*, EncodedJSValue baseObject, EncodedJSValue value, PropertyName);
-using PutValueFuncWithPtr = bool (JIT_OPERATION_ATTRIBUTES*)(JSGlobalObject*, EncodedJSValue baseObject, EncodedJSValue value, PropertyName, void*);
+using PutValueFunc = TypedFunctionPtr<PutValueFuncPtrTag, bool, JSGlobalObject*, EncodedJSValue, EncodedJSValue, PropertyName>;
```

Comment:
From what I said in slack, but want to put it into code review, too. Would be nice if we templatized `TypedFunctionPtr` like we do `Function`, so this LOC can become more readable by writing:
`TypedFunctionPtr<PutValueFuncPtrTag, bool(JSGlobalObject*, EncodedJSValue, EncodedJSValue, PropertyName)>`

Reply:
Done.